common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -465,7 +470,11 @@ public static void copyMemory(long srcAddr, byte[] dst, int dstIndex, long lengt
      * consumer (one thread!).
      */
     public static <T> Queue<T> newMpscQueue() {
-        return new MpscLinkedQueue<T>();
+        if (hasUnsafe()) {
+            return JAVA_VERSION < 8 ? new MpscLinkedQueue7<T>() : new MpscLinkedQueue8<T>();
```

Comment:
Nope java 5... That said we can just use CLQ for java5

> Am 28.03.2016 um 18:39 schrieb Guido Medina <notifications@github.com>:
> 
> In common/src/main/java/io/netty/util/internal/PlatformDependent.java:
> 
> > @@ -465,7 +470,11 @@ public static void copyMemory(long srcAddr, byte[] dst, int dstIndex, long lengt
> >       * consumer (one thread!).
> >       */
> >      public static <T> Queue<T> newMpscQueue() {
> > -        return new MpscLinkedQueue<T>();
> > +        if (hasUnsafe()) {
> > +            return JAVA_VERSION < 8 ? new MpscLinkedQueue7<T>() : new MpscLinkedQueue8<T>();
> Isn't Netty 4.0+ for JDK 6? JCtools minimal requirement is JDK 6
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>